docs/developer-resources.md

```diff
@@ -0,0 +1,346 @@
+         ```bash
+         cd client
+         npm install
+         ```
+
+         Then, run the `start` script and in a few seconds a tab should open in your browser automatically at `localhost:3000`
+
+   2. Otherwise, clone the repo as normal and set it up in your editor
+
+      ```bash
+      git clone https://github.com/team-monumental/monuments-and-memorials.git
+      ```
+
+      1. Run the Gradle build. This should download all the dependencies.
+
+         ```bash
+         ./gradlew build
+         ```
+
+      2. Start the server
+
+         ```bash
+         java -jar build/libs/monuments-and-memorials-0.0.1-SNAPSHOT.jar
+         ```
+
+      3. Run the npm install and start the React server.
+
+         ```bash
+         cd client
+         npm install
+         npm start
+         ```
+
+      4. Go to http://localhost:3000 if a browser tab does not open automatically
+
+4. Verify that everything's working
+
+   1. At http://localhost:3000, click around and make sure the UI looks the same as https://monuments.us.org/ (without any data).
+
+   2. Click the sign up button and try to sign up. It should succeed. Now, connect to postgres and make yourself an admin user.
+
+      ```bash
+      psql postgres postgres
+      ```
+
+      ```plsql
+      UPDATE "user" SET role = 'ADMIN' RETURNING *;
+      ```
+
+      You should see information about the user you just created returned.
+
+   3. In your browser, log out and log back in. You should now be able to access the admin panel.
+
+5. You can get some data into your database for testing by doing a bulk create from the admin panel using one of the initial data dumps from Dr. Decker's classes. The most up to date version in the Google Drive should upload without an warnings or errors.
+
+## Deploying
+
+1. Complete **Local Setup** above
+
+2. Checkout the version of the application you want to deploy (this should always be master)
+
+   ```bash
+   git fetch origin
+   git checkout -b master origin/master 
+   git pull
+   ```
+
+3. Add the VM as a git remote named `deploy`
+
+   ```bash
+   git remote add deploy ubuntu@monuments.us.org:/home/monumental/monuments-and-memorials 
+   ```
+
+4. Deploy
+
+   ```bash
+   git push deploy master 
+   ```
+
+   1. If you're pushing from a branch other than master,
+
+      ```bash
+      git push deploy my-branch:master
+      ```
+
+      Please note that if you do this, you will need to force push in the future to get the branch back to the real state of master.
+
+## Project Structure
+
+To get a good sense of how the project works, please check out the documentation in the Google Drive folder. The architecture diagram and document are way out of date and I don't have time to update them so I'm just going to brain dump here a bit.
+
+### Database Tier
+
+#### PostgreSQL
+
+Database. Pretty straightforward.
+
+##### pgtrgm
+
+PostgreSQL extension responsible for word matching, which is what you see in the first search box on the site.
```

Comment:
Maybe add that it allows us to do the fuzziness searching we do on titles, artists, descriptions, etc